lib/mix/tasks/gen/ash.gen.resource.ex

```diff
@@ -416,6 +416,63 @@ if Code.ensure_loaded?(Igniter) do
     defp add_relationships_to_resource(igniter, resource, options) do
       Enum.reduce(options[:relationship] || [], igniter, fn relationship, igniter ->
         case String.split(relationship, ":") do
+          ["many_to_many", name, through, destination | modifiers] ->
+            cond do
+              !String.contains?(through, ".") ->
+                raise ArgumentError, """
```

Comment:
We don't do this validation for other relationship types, let's not do it here either. It's up to the user really. We can potentially match it as a regex instead if we want, but if we do we should do it everywhere. Technically just `Foo` is a valid module name, so it's not sufficient to check for a `.`. My vote is to leave it out for now.